packages/expo-secure-store/android/src/main/java/expo/modules/securestore/SecureStoreModule.java

```diff
@@ -396,18 +400,18 @@ public JSONObject createEncryptedItem(String plaintextValue, KeyStore keyStore,
       SecretKey secretKey = secretKeyEntry.getSecretKey();
       Cipher cipher = Cipher.getInstance(AES_CIPHER);
       cipher.init(Cipher.ENCRYPT_MODE, secretKey);
+      GCMParameterSpec gcmSpec = cipher.getParameters().getParameterSpec(GCMParameterSpec.class);
 
-      return createEncryptedItem(plaintextValue, cipher);
+      return createEncryptedItem(plaintextValue, cipher, gcmSpec);
     }
 
-    /* package */ JSONObject createEncryptedItem(String plaintextValue, Cipher cipher) throws
+    /* package */ JSONObject createEncryptedItem(String plaintextValue, Cipher cipher, GCMParameterSpec gcmSpec) throws
         GeneralSecurityException, JSONException {
 
       byte[] plaintextBytes = plaintextValue.getBytes(StandardCharsets.UTF_8);
       byte[] ciphertextBytes = cipher.doFinal(plaintextBytes);
       String ciphertext = Base64.encodeToString(ciphertextBytes, Base64.DEFAULT);
 
-      GCMParameterSpec gcmSpec = cipher.getParameters().getParameterSpec(GCMParameterSpec.class);
```

Comment:
This ensures we use the same spec that the cipher uses -- it would be better to keep this if possible, and if we remove this I think we need to add an assertion that the cipher's algorithm matches the spec object that is passed in.

Reply:
On Android 23+ it works exactly as it did (we use `getParameterSpec()`, see https://github.com/expo/expo/pull/6464/files#diff-c4fb6019b00e115649c6aafbcb5d3d82R405).

The change had to be made for Androids below 23, which seem to ship with a faulty BouncyCastle version which tries to instantiate `GCMParameterSpec` with an invalid constructor (see [fixing commit](https://github.com/bcgit/bc-java/commit/507c3917c0c469d10b9f033ad641c1da195e2039#diff-c90a59e823805b6c0dcfeaf7bae65f53)).

I think `Cipher` would [throw an exception](https://android.googlesource.com/platform/libcore/+/d416195/luni/src/main/java/javax/crypto/Cipher.java#649) if it couldn't fulfill the `GCMParameterSpec` we provide him in `aesCipher.init(Cipher.ENCRYPT_MODE, secretKey, gcmSpec);`, so I don't believe that the assertion will ever fail, but I have created such sanity-checking path in code. 🙂

Reply:
When you approve of these changes I will backport the changes to older SDKs. (reminder for myself)